### PR TITLE
test: added tests to ensure we enqueueing all not failing messages

### PR DESCRIPTION
**Pull request by MadsDue**

### CODE REVIEW

File: source/B2BApi/Functions/EnqueueMessages/Activities/EnqueueEnergyResultsForBalanceResponsiblesActivity.cs [hunk 1]
```diff
@@ -94,30 +93,33 @@ await scopedOutgoingMessagesClient
                         numberOfFailedResults++;
                         _logger.LogWarning(
                             ex,
-                            "Enqueue and commit failed for energy result, query type: {QueryType}, external id: {ExternalId}, calculation id: {CalculationId}, event id: {EventId}",
+                            "Enqueue and commit failed for calculation result, query type: {QueryType}, external id: {ExternalId}, calculation id: {CalculationId}, event id: {EventId}",
```
Comment (lasrinnil) [inline source/B2BApi/Functions/EnqueueMessages/Activities/EnqueueEnergyResultsForBalanceResponsiblesActivity.cs:96]: `EnergyResultPerBalanceResponsiblePerGridAreaQuery` only produces energy results?

File: source/B2BApi.AppTests/Functions/EnqueueMessages/EnqueueMessagesOrchestrationTests.cs [hunk 2]
```diff
@@ -362,6 +353,130 @@ public async Task Given_DatabricksHasNoData_When_CalculationCompletedEventIsHand
...
+                        (item.Value<string>("FunctionName"), (string?)item.Value<dynamic>("FailureDetails")?.ErrorMessage))
+                    .ToList();
+
+                var containsExpectedHistory = expectedHistory.Intersect(activities).Count() == expectedHistory.Count();
+                return containsExpectedHistory;
+            },
+            TimeSpan.FromSeconds(30),
+            delay: TimeSpan.FromSeconds(5));
+
+        isExpected.Should().BeTrue("because we expect the actual history to contain the expected history");
```
Comment (lasrinnil) [inline source/B2BApi.AppTests/Functions/EnqueueMessages/EnqueueMessagesOrchestrationTests.cs:415]: Should we ensure that the actual messages are created through inspecting the `OutgoingMessages`?

File: source/B2BApi.AppTests/Functions/EnqueueMessages/EnqueueMessagesOrchestrationTests.cs [hunk 3]
```diff
@@ -111,10 +111,7 @@ await act.Should()
     public async Task Given_CalculationOrchestrationId_When_CalculationCompletedEventForBalanceFixingIsHandled_Then_OrchestrationCompletesWithExpectedServiceBusMessage()
     {
         // Arrange
-        Fixture.EnsureAppHostUsesFeatureFlagValue(
-            enableCalculationCompletedEvent: true,
-            enableCalculationCompletedEventForBalanceFixing: true,
-            enableCalculationCompletedEventForWholesaleFixing: true);
+        EnableCalculationCompletedEvent();
```
Comment (lasrinnil) [inline source/B2BApi.AppTests/Functions/EnqueueMessages/EnqueueMessagesOrchestrationTests.cs:114]:
```suggestion
        EnableEnqueueMessagesOrchestration();
```

File: source/B2BApi/Functions/EnqueueMessages/Activities/EnqueueEnergyResultsForBalanceResponsiblesAndEnergySuppliersActivity.cs [hunk 4]
```diff
@@ -59,29 +59,31 @@ public class EnqueueEnergyResultsForBalanceResponsiblesAndEnergySuppliersActivit
...
-
             // Only log databricks query time if it took more than 1 second
             if (databricksStopwatch.Elapsed > TimeSpan.FromSeconds(1))
             {
                 _logger.LogInformation(
                     "Retrieved energy result from databricks, elapsed time: {ElapsedTime}, type: {QueryType}, external id: {ExternalId}, calculation id: {CalculationId}, event id: {EventId}",
                     databricksStopwatch.Elapsed,
                     query.GetType().Name,
-                    queryResult.Result!.ExternalId.Value,
+                    queryResult.Result?.ExternalId.Value,
```
Comment (lasrinnil) [inline source/B2BApi/Functions/EnqueueMessages/Activities/EnqueueEnergyResultsForBalanceResponsiblesAndEnergySuppliersActivity.cs:69]: It is out of context of this PR. But should we not make sure that a results i present from the enumerator before we proceed? On line 85 we even have a supressing that the result could be `null`?

File: source/IntegrationTests/Behaviours/IntegrationEvents/TestData/EnergyResultPerBrpGridAreaDescription.cs [hunk 5]
```diff
@@ -28,7 +28,10 @@ public class EnergyResultPerBrpGridAreaDescription
     : TestDataDescription
 {
     public EnergyResultPerBrpGridAreaDescription()
-        : base("balance_fixing_01-11-2022_01-12-2022_ga_543_per_brp_ga_v1.csv")
+        : base(
+        "balance_fixing_01-11-2022_01-12-2022_ga_543_per_brp_ga_v1.csv",
+        //Metering point type on row 218 is invalid (first row on a result set)
```
Comment (lasrinnil) [inline source/IntegrationTests/Behaviours/IntegrationEvents/TestData/EnergyResultPerBrpGridAreaDescription.cs:33]: How is the row invalid?

**Review by lasrinnil — APPROVED** [review]
Great work !

File: source/B2BApi.AppTests/Functions/EnqueueMessages/EnqueueMessagesOrchestrationTests.cs (hunk 2 above)
Comment (MadsDue) [reply source/B2BApi.AppTests/Functions/EnqueueMessages/EnqueueMessagesOrchestrationTests.cs:415]: This tests is more focusing on the behavior of the orchestration and the activities.

File: source/B2BApi/Functions/EnqueueMessages/Activities/EnqueueEnergyResultsForBalanceResponsiblesAndEnergySuppliersActivity.cs (hunk 4 above)
Comment (MadsDue) [reply source/B2BApi/Functions/EnqueueMessages/Activities/EnqueueEnergyResultsForBalanceResponsiblesAndEnergySuppliersActivity.cs:69]: it happens already on line 74 :)

Comment (MadsDue) [reply source/B2BApi/Functions/EnqueueMessages/Activities/EnqueueEnergyResultsForBalanceResponsiblesAndEnergySuppliersActivity.cs:69]: This log statement purpose as I understands it, is to log the databricks query time even if the query doesn't success

**Pull request merged**